### PR TITLE
set default for nil offset

### PR DESCRIPTION
**Pull request by ryabrody**

### CODE REVIEW

File: lib/lhs/data.rb [hunk 1]
```diff
@@ -60,7 +60,7 @@ def respond_to_missing?(name, include_all = false)
   private
 
   def collection_proxy?(input)
-    !! (input.is_a?(Hash) && input[items_key]) || input.is_a?(Array) || _raw.is_a?(Array)
+    !! (input.is_a?(Hash) && (_record && input[items_key])) || input.is_a?(Array) || _raw.is_a?(Array)
```
Comment (pape-io) [inline lib/lhs/data.rb]: We might have to reconsider this one. Now he fallbacks to item proxy, even though it's a collection. As the only problem here was the delegation we might raise the default value of items key up to its delegator.

File: lib/lhs/concerns/record/all.rb [hunk 2]
```diff
@@ -16,12 +16,16 @@ module ClassMethods
...
         limit = params[limit_key] || DEFAULT_LIMIT
         data = request(params: params.merge(limit_key => limit))
-        request_all_the_rest(data, params) if data._raw.is_a?(Hash) && data._raw[total_key]
+        request_all_the_rest(data, params) if is_paginated?(data._raw)
         data._record.new(LHS::Data.new(data, nil, self))
       end
 
       private
 
+      def is_paginated?(data_raw)
```
Comment (pape-io) [inline lib/lhs/concerns/record/all.rb]: I would just call it `raw`

**Comment by pape-io** [comment]
1 comment the rest is 👍

**Comment by ryabrody** [comment]
@pape-io done.

**Pull request merged**